### DISCUSSION TITLE
clang-format not working when file saved in remote development

### DISCUSSION THREAD

**Discussion by bharadwajy** [microsoft/vscode-cpptools#8732] [Q&A] [answered]
I am using VSCode running on MacOS for remote development on a Ubuntu machine.

I have `/home/my/workspace/subdir/.clang-format`. I turned on `Editor: Format On Save`

I edit and save `/home/my/workspace/subdir/anotherdir/file.cpp`. Upon saving the file, the file is not formatted.

Any suggestions to address the issue?

Thanks!

**Comment by bobbrow** [comment, marked as answer]
Can you [Enable Debug Logging](https://code.visualstudio.com/docs/cpp/enable-logging-cpp), try again, and then share the logging with us?

**Reply by bharadwajy** [reply]
Thanks for your reply. 

It appears that the `.clang-format` I am using has an error. I was able to get the `Format On Save` feature work by modifying the `.clang-format`.

**Reply by bobbrow** [reply]
That's great news. Let us know if you need anything else.
